app/src/main/java/com/example/egida/domain/useCase/scoreBall/UseCaseScoreBalImpl.kt

```diff
@@ -0,0 +1,138 @@
+package com.example.egida.domain.useCase.scoreBall
+
+import android.util.Log
+import com.example.egida.Dependencies
+import com.example.egida.domain.entity.UserDatabase
+import com.example.egida.domain.useCase.day.DayUseCase
+import com.example.egida.domain.useCase.userDatabase.UserDatabaseUseCase
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.launch
+
+class UseCaseScoreBalImpl : UseCaseScoreBal {
+    companion object {
+        const val TAG = "UseCaseScoreBalImpl"
+        const val steep: Int = 35
+    }
+
+    private val dayUseCase: DayUseCase by lazy { Dependencies.dayUseCase() }
+    private val userDatabaseUseCase: UserDatabaseUseCase by lazy { Dependencies.userDatabaseUseCase() }
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+    var day = dayUseCase.day
+        .shareIn(scope, started = SharingStarted.Eagerly, replay = 1)
+    var user: SharedFlow<UserDatabase> = userDatabaseUseCase.databaseUser
+        .shareIn(scope, started = SharingStarted.Lazily, replay = 1)
+    private var bodyMassIndex: Int = 0
+    private var waterNorm: Double = 0.00
+    private var weight: Int = 0
+    private var height: Int = 0
+    private var userDatabase: UserDatabase = UserDatabase()
+
+    private fun calculationBodyMassIndex(weight: Int, height: Int): Int {
+
+        val doubleHeight = ((height.toDouble() / 100) * (height.toDouble() / 100))
+        bodyMassIndex = (weight / doubleHeight).toInt()
+        Log.d(TAG, " bodyMassIndex $bodyMassIndex")
+        return bodyMassIndex
+    }
+
+    private fun calculationWaterNorm(weight: Int): Double {
+
+        when (weight) {
+            in 0..9 -> waterNorm = 0.25
```

Comment:
There are a lot of magic constants in this file. It's better to move them to the constants. You can leave it for now, while app code work is in progress, but create at least a comment to replace this constants.